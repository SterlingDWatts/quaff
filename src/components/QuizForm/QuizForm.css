.QuizForm {
  height: 100%;
  padding: 8px;
}

.QuizForm fieldset {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.QuizForm input {
  margin: 3px 5px 0 0;
}
