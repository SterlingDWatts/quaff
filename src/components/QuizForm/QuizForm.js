import React, { Component } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheck, faTimes } from "@fortawesome/free-solid-svg-icons";
import QuizContext from "../../contexts/QuizContext";
import "./QuizForm.css";

class QuizForm extends Component {
  static contextType = QuizContext;

  renderAnswers = (answers) => {
    return answers.map((answer) => {
      let correct = answer.correct ? faCheck : faTimes;
      let icon = this.context.showAnswer ? (
        <FontAwesomeIcon icon={correct} />
      ) : (
        <></>
      );
      return (
        <div className="QuizForm__label_group" key={answer.id}>
          <input
            type="radio"
            name="answer"
            id={`ans_${answer.id}`}
            value={answer.id}
            required
            disabled={this.context.showAnswer}
          />
          <label htmlFor={`ans_${answer.id}`}>
            {answer.answer} {icon}
          </label>
        </div>
      );
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const selected = Number(event.target.answer.value);
    if (!this.context.showAnswer) {
      this.context.revealAnswer(selected);
    } else {
      this.context.updateProgress();
    }
  };

  render() {
    const { quiz, progress, showAnswer, error } = this.context;
    let content;
    if (error) {
      content = <div className="QuizForm--error">Error</div>;
    } else if (quiz.length < 1) {
      content = <div className="QuizForm--loading"></div>;
    } else {
      const question = quiz[progress];
      const answers = this.renderAnswers(question.answers);
      content = (
        <fieldset>
          <legend className="QuizForm__question">{question.question}</legend>
          {answers}
          <button className="QuizForm__button" type="submit">
            {showAnswer ? "Next" : "Submit"}
          </button>
        </fieldset>
      );
    }
    return (
      <form className="QuizForm" onSubmit={this.handleSubmit}>
        {content}
      </form>
    );
  }
}

export default QuizForm;
