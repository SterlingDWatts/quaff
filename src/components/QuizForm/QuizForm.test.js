import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import QuizForm from "./QuizForm";

describe("QuizForm Component", () => {
  it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <QuizForm />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <QuizForm />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
