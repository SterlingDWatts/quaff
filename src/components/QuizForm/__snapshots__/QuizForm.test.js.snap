// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizForm Component renders the UI as expected 1`] = `
<form
  className="QuizForm"
  onSubmit={[Function]}
>
  <div
    className="QuizForm--loading"
  />
</form>
`;
